Improve WTF_ALLOW_UNSAFE_BUFFER_USAGE macro usage

cdumez opened the pull request:
#### 73bef3bfa1a993dd2e9abc5b8c64f018bfb6f4f8
<pre>
Improve WTF_ALLOW_UNSAFE_BUFFER_USAGE macro usage
<a href="https://bugs.webkit.org/show_bug.cgi?id=285235">https://bugs.webkit.org/show_bug.cgi?id=285235</a>

Reviewed by Darin Adler.

Use WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_* where possible and
add comments for port-specific WTF_ALLOW_UNSAFE_BUFFER_USAGE to
facilitate identification.

* Source/WTF/wtf/glib/ChassisType.cpp:
(WTF::readMachineInfoChassisType):
* Source/WTF/wtf/glib/SocketConnection.cpp:
(WTF::SocketConnection::read):
(WTF::SocketConnection::readMessage):
(WTF::SocketConnection::sendMessage):
(WTF::SocketConnection::write):
* Source/WebCore/Modules/mediastream/RTCIceCandidateFields.cpp:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCCertificateGenerator.cpp:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCDataChannelHandler.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCObservers.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpTransformBackend.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpTransformableFrame.cpp:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCUtils.cpp:
* Source/WebCore/PAL/pal/PALSwift.h:
* Source/WebCore/PAL/pal/spi/cocoa/CoreCryptoSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/DataDetectorsCoreSPI.h:
* Source/WebCore/platform/cocoa/SharedVideoFrameInfo.mm:
* Source/WebCore/platform/graphics/Color.h:
* Source/WebCore/platform/graphics/FontPlatformData.h:
* Source/WebCore/platform/graphics/FontRenderOptions.h:
* Source/WebCore/platform/graphics/PlatformColorSpace.h:
* Source/WebCore/platform/graphics/PlatformImage.h:
* Source/WebCore/platform/graphics/cv/VideoFrameCV.mm:
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::getEGLConfig):
(WebCore::GLContext::createContextForEGLVersion):
(WebCore::GLContext::isExtensionSupported):
* Source/WebCore/platform/graphics/skia/PatternSkia.cpp:
* Source/WebCore/platform/libwebrtc/LibWebRTCVPXVideoEncoder.cpp:
* Source/WebCore/platform/mediastream/RealtimeIncomingAudioSource.h:
* Source/WebCore/platform/mediastream/RealtimeIncomingVideoSource.h:
* Source/WebCore/platform/mediastream/RealtimeOutgoingAudioSource.h:
* Source/WebCore/platform/mediastream/RealtimeOutgoingVideoSource.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCAudioModule.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCDTMFSenderBackend.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCDav1dDecoder.cpp:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCDav1dDecoder.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.cpp:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
* Source/WebCore/platform/mediastream/libwebrtc/VideoFrameLibWebRTC.h:
* Source/WebCore/platform/video-codecs/cocoa/WebRTCVideoDecoder.mm:
* Source/WebCore/testing/MockLibWebRTCPeerConnection.h:
* Source/WebKit/UIProcess/gtk/DisplayX11.cpp:
(WebKit::Display::accessibilityBusAddressX11 const):
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.h:

Canonical link: <a href="https://commits.webkit.org/288326@main">https://commits.webkit.org/288326@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/08cb1db46c56d503c7f9ef4aee158986f86133d5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82539 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/2195 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/36642 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/87664 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/33600 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/84637 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/2266 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/10095 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64326 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/22094 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85600 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/1633 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/75093 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44603 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1528 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/29281 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/32634 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/75527 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/72818 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/29935 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/89027 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/81591 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9837 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/7103 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/72735 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/10065 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70907 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71951 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/16093 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/15093 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/1225 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12806 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9790 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/15311 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/103998 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9664 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25224 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13130 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/11434 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/glib/ChassisType.cpp:
```diff
@@ -43,7 +43,7 @@ static std::optional<ChassisType> readMachineInfoChassisType()
 
     GUniquePtr<char*> split(g_strsplit(buffer.get(), "\n", -1));
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GLib port.
```
Not sure this speaks for itself; I don't fully understand why emphasizing which port this code is specific to belongs in a comment on the macro invocation line. I don't have an objection to it if it's helpful but I don't know the rationale.

cdumez replied:
I'm following the pattern previously established by @geoffreygaren. The idea is to help identify remaining `WTF_ALLOW_UNSAFE_BUFFER_USAGE` usage that impacts Apple ports (vs not) when grep'ing

darinadler commented on Source/WebCore/platform/graphics/egl/GLContext.cpp:
```diff
@@ -80,7 +80,7 @@ bool GLContext::getEGLConfig(PlatformDisplay& platformDisplay, EGLConfig* config
     }
 
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // Non-Apple ports.
```
Seems a little Apple-centric to write a comment like this. Is there a more principled way to do this, and still achieve the goal of being able to drive these down to zero for the Apple platforms?

cdumez replied:
Ideally I'd specify which ports are using this code. I'll try and figure this out.

darinadler commented on Source/WebCore/platform/graphics/skia/PatternSkia.cpp:
```diff
@@ -31,9 +31,9 @@
+WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_BEGIN
 #include <skia/core/SkMatrix.h>
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_END
```
Seems to me that we should move this out of the big paragraph of includes when it needs to be wrapped with macros like this. But I suppose we shouldn't mess with it if it's working.

cdumez replied:
Ok.

cdumez committed:
Improve WTF_ALLOW_UNSAFE_BUFFER_USAGE macro usage
https://bugs.webkit.org/show_bug.cgi?id=285235

Reviewed by Darin Adler.

Use WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_* where possible and
add comments for port-specific WTF_ALLOW_UNSAFE_BUFFER_USAGE to
facilitate identification.

* Source/WTF/wtf/glib/ChassisType.cpp:
(WTF::readMachineInfoChassisType):
* Source/WTF/wtf/glib/SocketConnection.cpp:
(WTF::SocketConnection::read):
(WTF::SocketConnection::readMessage):
(WTF::SocketConnection::sendMessage):
(WTF::SocketConnection::write):
* Source/WebCore/Modules/mediastream/RTCIceCandidateFields.cpp:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCCertificateGenerator.cpp:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCDataChannelHandler.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCObservers.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpTransformBackend.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpTransformableFrame.cpp:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCStatsCollector.h:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCUtils.cpp:
* Source/WebCore/PAL/pal/PALSwift.h:
* Source/WebCore/PAL/pal/spi/cocoa/CoreCryptoSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/DataDetectorsCoreSPI.h:
* Source/WebCore/platform/cocoa/SharedVideoFrameInfo.mm:
* Source/WebCore/platform/graphics/Color.h:
* Source/WebCore/platform/graphics/FontPlatformData.h:
* Source/WebCore/platform/graphics/FontRenderOptions.h:
* Source/WebCore/platform/graphics/PlatformColorSpace.h:
* Source/WebCore/platform/graphics/PlatformImage.h:
* Source/WebCore/platform/graphics/cv/VideoFrameCV.mm:
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::getEGLConfig):
(WebCore::GLContext::createContextForEGLVersion):
(WebCore::GLContext::isExtensionSupported):
* Source/WebCore/platform/graphics/skia/PatternSkia.cpp:
* Source/WebCore/platform/libwebrtc/LibWebRTCVPXVideoEncoder.cpp:
* Source/WebCore/platform/mediastream/RealtimeIncomingAudioSource.h:
* Source/WebCore/platform/mediastream/RealtimeIncomingVideoSource.h:
* Source/WebCore/platform/mediastream/RealtimeOutgoingAudioSource.h:
* Source/WebCore/platform/mediastream/RealtimeOutgoingVideoSource.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCAudioModule.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCDTMFSenderBackend.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCDav1dDecoder.cpp:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCDav1dDecoder.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.cpp:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
* Source/WebCore/platform/mediastream/libwebrtc/VideoFrameLibWebRTC.h:
* Source/WebCore/platform/video-codecs/cocoa/WebRTCVideoDecoder.mm:
* Source/WebCore/testing/MockLibWebRTCPeerConnection.h:
* Source/WebKit/UIProcess/gtk/DisplayX11.cpp:
(WebKit::Display::accessibilityBusAddressX11 const):
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.h:

Canonical link: https://commits.webkit.org/288326@main
